meteor/server/api/blueprints/context/context.ts

```diff
@@ -594,91 +583,109 @@ export class AsRunEventContext extends RundownContext implements IAsRunEventCont
 			).fetch()
 		)
 	}
-	/** Get all segments in this rundown */
-	getSegments(): Array<IBlueprintSegmentDB> {
-		return unprotectObjectArray(this._rundown.getSegments())
+
+	formatDateAsTimecode(time: number): string {
+		check(time, Number)
+		return formatDateAsTimecode(new Date(time))
 	}
-	/**
-	 * Returns a segment
-	 * @param segmentId Id of segment to fetch. If is omitted, return the segment related to this AsRunEvent
-	 */
-	getSegment(segmentId?: string): IBlueprintSegmentDB | undefined {
-		segmentId = segmentId || this.asRunEvent.segmentId
-		check(segmentId, String)
-		if (segmentId) {
-			return unprotectObject(
-				this._rundown.getSegments({
-					_id: protectString(segmentId),
-				})[0]
-			)
-		}
+	formatDurationAsTimecode(time: number): string {
+		check(time, Number)
+		return formatDurationAsTimecode(time)
 	}
-	/** Get all parts in this rundown */
-	getParts(): Array<IBlueprintPartDB> {
-		return unprotectObjectArray(this._rundown.getParts())
+}
+
+export class RundownTimingEventContext extends RundownDataChangedEventContext implements IRundownTimingEventContext {
+	readonly previousPart: Readonly<IBlueprintPartInstance<unknown>> | undefined
+	private readonly _part: PartInstance
+	readonly nextPart: Readonly<IBlueprintPartInstance<unknown>> | undefined
+
+	public get part(): Readonly<IBlueprintPartInstance<unknown>> {
```

Comment:
Question: Should this be named currentPart, to be constistent with previous & next?

Reply:
I did wonder that, but this is consistent with some other interfaces here.
But yeah I think I will change this to currentPart for better symmetry within this interface